layers/+lang/c-c++/funcs.el

```diff
@@ -43,7 +43,10 @@
   ;; currently DAP is only available using LSP
   (pcase c-c++-backend
     ('lsp-clangd (spacemacs//c-c++-setup-lsp-dap))
-    ('lsp-ccls (spacemacs//c-c++-setup-lsp-dap))))
+    ('lsp-ccls (spacemacs//c-c++-setup-lsp-dap))
+    (_ (user-error "Unknown `c-c++-backend': %s" c-c++-backend)))
+  (setq-local spacemacs--dap-supported-modes
+              '(c-mode c++-mode)))
```

Comment:
This should not be necessary, the list of supported dap modes is globally set among all layers, a local version of this variable could cause havoc, i.e. go into a python file, swtich to c++-mode and switch back your buffer will not longer have dap support for python available.

Instead this list should be extended in the `pre-init-dap` section.